module_utils/network_lsr/argument_validator.py

```diff
@@ -779,15 +779,27 @@ def _validate_post(self, value, name, result):
                     name,
                     "missing 'table' for the routing rule",
                 )
-
-        if result["from"] is not None:
+        # `from 0.0.0.0/0` means from all IPv4 addresses
+        # `from ::/0` means from all IPv6 addresses
+        # In NM, if `from` property is not specified in a routing rule, NM
+        # still appends `from 0.0.0.0/0` or `from ::/0` to the rule
+        if result["from"] is not None and result["from"]["address"] not in [
+            "0.0.0.0",
```

Comment:
Above you say "`from 0.0.0.0/0` means from all IPv4 addresses" but here the value is `"0.0.0.0"` - which one is correct?

Reply:
Is this true for all versions of NM in el7, el8, el9, and Fedora?

Reply:
Is there some corresponding change that will need to be made in the nmstate implementation?

Reply:
> Is there some corresponding change that will need to be made in the nmstate implementation?

There are some corresponding changes that will need to be made in NM, and very likely the update will only include in the NM newest release. And originally, the user requested to specify `from ::/0` in the routing rule, but the user can achieve the same effect by not specifying `from ::/0` in the routing rule (NM will automatically append `from ::/0` to the routing rule). Therefore, I will request that user to not specify `from ::/0` when using the  network role.